### ISSUE TITLE
Build warning in `nolibc/string.c`

### ISSUE THREAD

**Issue by razvand** [unikraft/unikraft#543]
### Describe the bug

A build warning related to `strncat` appears when using `nolibc/string.c`:

```
/home/razvan/usoc/unikraft/lib/nolibc/string.c: In function ‘strncat’:
/home/razvan/usoc/unikraft/lib/nolibc/string.c:696:5: warning: nonnull argument ‘src’ compared to NULL [-Wnonnull-compare]
  if (src != NULL) {   
```

**Hackathon points**: 15

### Steps to reproduce

Build an application that uses `nolibc`.

### Expected behavior

There should be no build warnings.

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

**Comment by razvand** [comment]
This only happens on GCC <= 9. For GCC >= 10, the warning isn't shown.

**Comment by van-ema** [comment]
Emanuele Vannacci <e.vannacci@vu.nl> working on this

**Comment by razvand** [comment]
> Emanuele Vannacci [e.vannacci@vu.nl](mailto:e.vannacci@vu.nl) working on this

@van-ema , @mariamosn has fixed this in #546

**Comment by razvand** [comment]
Fixed in #546
